service/worker/workerdeployment/replaytester/worker/worker.go

```diff
@@ -151,6 +163,42 @@ func main() {
 
 }
 
+func saveHistories(c client.Client, deploymentName, buildId string) {
+	deploymentWFEvents := c.GetWorkflowHistory(
+		context.Background(),
+		worker_versioning.GenerateDeploymentWorkflowID(deploymentName),
+		"",
+		false,
+		enumspb.HISTORY_EVENT_FILTER_TYPE_ALL_EVENT,
+	)
+	deploymentWFHistory := make([]*history.HistoryEvent, 0)
+	for deploymentWFEvents.HasNext() {
+		event, err := deploymentWFEvents.Next()
+		if err != nil {
+			panic(err)
+		}
+		deploymentWFHistory = append(deploymentWFHistory, event)
+	}
+	// TODO: write history to json file
+
+	versionWFHistory := make([]*history.HistoryEvent, 0)
+	versionWFEvents := c.GetWorkflowHistory(
+		context.Background(),
+		worker_versioning.GenerateVersionWorkflowID(deploymentName, buildId),
```

Comment:
you might actually need to get the run id of the previous workflow run to do this correctly? not sure